lib/utils/report.cjs

```diff
@@ -20,18 +20,23 @@ const constants = require('../constants.cjs');
  * @type {import('stylelint').Utils['report']}
  */
 function report(problem) {
-	const { node, index, endIndex, line, start, end, result, ruleName, word, ...rest } = problem;
+	const { node, index, endIndex, line, start, end, result, ruleName, word, fix, ...rest } = problem;
 	const {
 		disabledRanges,
 		quiet,
 		ruleSeverities,
 		config: { defaultSeverity = 'error', ignoreDisables } = {},
 		customMessages: { [ruleName]: message = rest.message } = {},
 		customUrls: { [ruleName]: customUrl } = {},
+		ruleMetadata: { [ruleName]: metadata } = {},
 	} = result.stylelint;
 	const { messageArgs = [], severity = ruleSeverities[ruleName] } = rest;
 	const ruleSeverity = validateTypes.isFunction(severity) ? severity(...messageArgs) || defaultSeverity : severity;
 
+	if (validateTypes.isFunction(fix) && metadata && !metadata.fixable) {
+		throw new Error(`The "${ruleName}" rule requires "fixable=true" metadata for autofix`);
+	}
```

Comment:
nit

You want this to be checked after line 41.
I don't know why someone would depend on it but that was the legacy behaviour.
i.e. if `(quiet && ruleSeverity !== 'error')` is truthy you wouldn't have errors (e.g. the startLine condition that throws is placed after)

Reply:
That makes sense. Changed via 1b40515c5. 👍🏼